src/components/DialogHeader/DialogHeader.tsx

```diff
@@ -0,0 +1,60 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import React, { ReactElement } from 'react';
+import { merge } from '@utilities/merge';
+import { IconCross } from '@foundation/Icon';
+import { Badge, BadgeProps } from '@components/Badge';
+
+export type DialogHeaderProps = {
+    title: string;
+    size?: DialogHeaderSize;
+    collapseBottom?: boolean;
+    close?: boolean;
+    onClose?: () => void;
+    icon?: ReactElement;
+    badge?: BadgeProps;
```

Comment:
Why not pass the `badge` prop as a component just like an `icon`? This introduces confusion when consuming components, sticking to one pattern removes it. And the better one IMO is by passing component.